src/storage/blocksstable/ob_micro_block_row_scanner.cpp

```diff
@@ -469,14 +469,21 @@ int ObMicroBlockRowScanner::init(const ObTableIterParam& param, ObTableAccessCon
                  projector,
                  storage::ObMultiVersionRowkeyHelpper::MVRC_NONE /*multi version col type*/))) {
     STORAGE_LOG(WARN, "Fail to init col map, ", K(ret));
+  } else if (OB_FAIL(obj_buf_.init(context.allocator_))) {
```

Comment:
context.allocator_'s lifecycle is not enough, in rescan cases, allocator_ will be released in each scan, use stmt_allocator_ instead